components/script/dom/workerglobalscope.rs

```diff
@@ -701,3 +707,9 @@ impl WorkerGlobalScope {
             .cancel_all_tasks_and_ignore_future_tasks();
     }
 }
+
+struct WorkerCspProcessor {}
```

Comment:
This means we will receive no violation reports when importScripts is blocked by CSP. Shouldn't we store a way to communicate with the WorkerGlobalScope here? I think a DomRoot will do the trick.

Reply:
Sure, I thought doing things incrementally here as I wasn't sure where this would be used yet. Added it and updated test expectations.